### PR TITLE
Convert decimals for ACA/KAR

### PR DESCRIPTION
**Pull request by zjb0807**
Closes: #1640

bodhi.js companion: https://github.com/AcalaNetwork/bodhi.js/pull/161

### CODE REVIEW

File: primitives/src/lib.rs [hunk 1]
```diff
@@ -177,3 +177,26 @@ pub enum ReserveIdentifier {
 pub type NFTBalance = u128;
 
 pub type CashYieldIndex = u128;
+
+/// Increase decimals
+pub fn convert_decimals_inc(b: Balance) -> Balance {
```
Comment (xlc) [inline primitives/src/lib.rs:182]: The name is a bit confusing. Maybe `convert_decimals_to_evm` and `convert_decimals_from_evm`

And comment needs to explain this convert decimal between native (12) and EVM (18) and therefore the `1_000_000` conversion.

File: primitives/src/lib.rs [hunk 2]
```diff
@@ -177,3 +177,26 @@ pub enum ReserveIdentifier {
...
+	if b.is_zero() {
+		return b;
+	}
+	b.saturating_mul(10u128.saturating_pow(6))
```
Comment (xlc) [inline primitives/src/lib.rs:186]:
```suggestion
	b.saturating_mul(1_000_000)
```

File: modules/evm/src/lib.rs [hunk 3]
```diff
@@ -866,23 +869,15 @@ pub mod module {
...
-				.checked_div(10u128.saturating_pow(num))
-				.expect("divisor is non-zero; qed"),
-		)
-	}
-
 	/// Get StorageDepositPerByte of actual decimals
 	pub fn get_storage_deposit_per_byte() -> BalanceOf<T> {
-		Self::truncate_zero(T::StorageDepositPerByte::get(), 6)
+		// StorageDepositPerByte decimals is 18, KAR/ACA decimals is 12, convert to 12 here.
+		BalanceOf::<T>::unique_saturated_from(
```
Comment (xlc) [inline modules/evm/src/lib.rs:875]: should make convert decimals generic to avoid those `unique_saturated_from`

File: modules/evm/src/lib.rs [hunk 4]
```diff
@@ -872,12 +871,9 @@ impl<T: Config> Pallet<T> {
...
 	pub fn get_storage_deposit_per_byte() -> BalanceOf<T> {
 		// StorageDepositPerByte decimals is 18, KAR/ACA decimals is 12, convert to 12 here.
-		BalanceOf::<T>::unique_saturated_from(
-			convert_decimals_dec(UniqueSaturatedInto::<u128>::unique_saturated_into(
-				T::StorageDepositPerByte::get(),
-			))
-			.expect("StorageDepositPerByte generate by convert_decimals_inc; qed"),
-		)
+		convert_decimals_from_evm::<BalanceOf<T>>(T::StorageDepositPerByte::get())
+			.expect("StorageDepositPerByte generate by convert_decimals_to_evm; qed")
```
Comment (xlc) [inline modules/evm/src/lib.rs:875]: could have an integrity_test to ensure this

**Review by xlc — APPROVED** [review]

**Review by ermalkaleci — APPROVED** [review]

**Pull request merged**